pkg/oktaapi/client_wrapper.go

```diff
@@ -0,0 +1,174 @@
+package oktaapi
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/okta/okta-sdk-golang/v2/okta"
+	"github.com/okta/okta-sdk-golang/v2/okta/query"
+	"go.uber.org/zap"
+
+	"github.com/cmsgov/easi-app/pkg/appcontext"
+	"github.com/cmsgov/easi-app/pkg/models"
+)
+
+// ClientWrapper is a wrapper around github.com/okta/okta-sdk-golang/v2/okta Client type.
+// The purpose of this package is to act as a drop-in replacement for the CEDAR LDAP API, so this package should implement the same
+// methods that the Client interface defined in package usersearch
+type ClientWrapper struct {
+	oktaClient *okta.Client
+}
+
+// NewClient creates a Client
+func NewClient(url string, token string) (*ClientWrapper, error) {
+	// TODO Do we need the "Context" response from okta.NewClient??
+	_, oktaClient, oktaClientErr := okta.NewClient(context.TODO(), okta.WithOrgUrl(url), okta.WithToken(token))
+	if oktaClientErr != nil {
+		return nil, oktaClientErr
+	}
+	return &ClientWrapper{ //TODO: implement the next function
+		oktaClient: oktaClient,
+	}, nil
+}
+
+// oktaUserResponse is used to marshal the JSON response from Okta into a struct
+type oktaUserResponse struct {
+	FirstName   string `json:"firstName"`
+	LastName    string `json:"lastName"`
+	DisplayName string `json:"displayName"`
+	Email       string `json:"email"`
+	Login       string `json:"login"`
+	SourceType  string `json:"SourceType"`
+}
+
+func (cw *ClientWrapper) parseOktaProfileResponse(ctx context.Context, profile *okta.UserProfile) (*oktaUserResponse, error) {
+	logger := appcontext.ZLogger(ctx)
+
+	// Create an okaUserProfile to return
+	parsedProfile := &oktaUserResponse{}
+
+	// Marshal the profile into a string so we can later unmarshal it into a struct
+	responseString, err := json.Marshal(profile)
+	if err != nil {
+		logger.Error("error marshalling okta response", zap.Error(err))
+		return nil, err
+	}
+
+	// Unmarshal the string into the oktaUserProfile type
+	err = json.Unmarshal(responseString, parsedProfile)
+	if err != nil {
+		logger.Error("error unmarshalling okta response", zap.Error(err))
+		return nil, err
+	}
+
+	return parsedProfile, nil
+}
+
+func (o *oktaUserResponse) toUserInfo() *models.UserInfo {
+	return &models.UserInfo{
+		CommonName: o.DisplayName,
+		Email:      models.NewEmailAddress(o.Email),
+		EuaUserID:  o.Login,
+		// DisplayName: o.DisplayName,
```

Comment:
Can we remove all this commented-out code that's not needed?

Reply:
Addressed in [6420f51](https://github.com/CMSgov/easi-app/pull/2046/commits/6420f51a448742a7f3a31ce0745e6ab4428c5c78)